### ISSUE TITLE
Failed to generate with oapi-codegen v2.3.0 due to openapi3.CircularReferenceCounter is undefined

### ISSUE THREAD

**Issue by 42milez** [42milez/go-oidc-expt#183]
Workaround:
To avoid the following error, use `github.com/oapi-codegen/oapi-codegen/v2 v2.3.1-0.20240709092903-309b11f85c73` .

```
# github.com/oapi-codegen/oapi-codegen/v2/pkg/util
../../../../../.go/pkg/mod/github.com/oapi-codegen/oapi-codegen/v2@v2.3.0/pkg/util/loader.go:24:45: undefined: openapi3.CircularReferenceCounter
../../../../../.go/pkg/mod/github.com/oapi-codegen/oapi-codegen/v2@v2.3.0/pkg/util/loader.go:26:12: undefined: openapi3.CircularReferenceCounter
../../../../../.go/pkg/mod/github.com/oapi-codegen/oapi-codegen/v2@v2.3.0/pkg/util/loader.go:33:12: undefined: openapi3.CircularReferenceCounter
app/idp/api/api.go:8: running "go": exit status 1
make: *** [gen] Error 1
```

**Comment by jamietanna** [comment]
Related: https://github.com/oapi-codegen/oapi-codegen/issues/1702

This is fixed as of v2.4.0

**Comment by 42milez** [comment]
I close this issue since this repository is going to be archived.
